src/ncp/ncp_base_ftd.cpp

```diff
@@ -592,6 +594,230 @@ void NcpBase::HandleTmfProxyStream(otMessage *aMessage, uint16_t aLocator, uint1
 }
 #endif // OPENTHREAD_ENABLE_TMF_PROXY
 
+otError NcpBase::DecodeOperationalDataset(otOperationalDataset &aDataset, const uint8_t **aTlvs, uint8_t *aTlvsLength)
+{
+    otError error = OT_ERROR_NONE;
+
+    memset(&aDataset, 0, sizeof(otOperationalDataset));
+
+    if (aTlvs != NULL)
+    {
+        *aTlvs = NULL;
+    }
+
+    if (aTlvsLength != NULL)
+    {
+        *aTlvsLength = 0;
+    }
+
+    while (!mDecoder.IsAllReadInStruct())
+    {
+        unsigned int propKey;
+
+        SuccessOrExit(error = mDecoder.OpenStruct());
+        SuccessOrExit(error = mDecoder.ReadUintPacked(propKey));
+
+        switch (static_cast<spinel_prop_key_t>(propKey))
+        {
+        case SPINEL_PROP_DATASET_ACTIVE_TIMESTAMP:
+            SuccessOrExit(error = mDecoder.ReadUint64(aDataset.mActiveTimestamp));
+            aDataset.mIsActiveTimestampSet = true;
+            break;
+
+        case SPINEL_PROP_DATASET_PENDING_TIMESTAMP:
+            SuccessOrExit(error = mDecoder.ReadUint64(aDataset.mPendingTimestamp));
+            aDataset.mIsPendingTimestampSet = true;
+            break;
+
+        case SPINEL_PROP_NET_MASTER_KEY:
+        {
+            const uint8_t *key;
+            uint16_t len;
+
+            SuccessOrExit(error = mDecoder.ReadData(key, len));
+            VerifyOrExit(len == OT_MASTER_KEY_SIZE, error = OT_ERROR_PARSE);
+            memcpy(aDataset.mMasterKey.m8, key, len);
+            aDataset.mIsMasterKeySet = true;
+            break;
+        }
+
+        case SPINEL_PROP_NET_NETWORK_NAME:
+        {
+            const char *name;
+            size_t len;
+
+            SuccessOrExit(error = mDecoder.ReadUtf8(name));
+            len = strlen(name);
+            VerifyOrExit(len <= OT_NETWORK_NAME_MAX_SIZE, error = OT_ERROR_INVALID_ARGS);
```

Comment:
You seem to use `OT_ERROR_INVALID_ARGS` and `OT_ERROR_PARSE` interchangeably here.  Is that intentional?

Reply:
No. Thanks. Changed this to return `OT_ERROR_INVALID_ARGS` if there is an incorrect value and `OT_ERROR_PARSE` only for parsing errors.